pkg/util/winutil/service.go

```diff
@@ -0,0 +1,241 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+// +build windows
+
+package winutil
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"time"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	defaultServiceCommandTimeout = 10
+)
+
+var (
+	modadvapi32               = windows.NewLazyDLL("advapi32.dll")
+	procEnumDependentServices = modadvapi32.NewProc("EnumDependentServicesW")
+)
+
+type enumServiceState uint32
+
+func OpenSCManager(desiredAccess uint32) (*mgr.Mgr, error) {
+	h, err := windows.OpenSCManager(nil, nil, desiredAccess)
+	if h == 0 {
+		return nil, err
+	}
+	return &mgr.Mgr{Handle: h}, err
+}
+
+func OpenService(manager *mgr.Mgr, serviceName string, desiredAccess uint32) (*mgr.Service, error) {
+	h, err := windows.OpenService(manager.Handle, windows.StringToUTF16Ptr(serviceName), desiredAccess)
+	if err != nil {
+		return nil, err
+	}
+	return &mgr.Service{Name: serviceName, Handle: h}, err
+}
+
+// Start serviceName via SCM
+// Does not block until service is started
+// https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicea#remarks
+func StartService(serviceName string) error {
+
+	m, err := OpenSCManager(windows.SC_MANAGER_CONNECT)
+	if err != nil {
+		return err
+	}
+	defer m.Disconnect()
+
+	s, err := OpenService(m, serviceName, windows.SERVICE_START)
+	if err != nil {
+		return fmt.Errorf("could not open service: %v", err)
+	}
+	defer s.Close()
```

Comment:
This section is used in many places, I wonder if it worth it to factored into separate function. It should save lot of lines

Reply:
exactly!

Reply:
I thought about it a little more - I'd prefer to keep it as-is for now because it isn't clear from the docs if we can safely disconnect the SCM handle and still use the service handle. Happy to revisit, but for this PR, I'm going to leave it.

cc @clarkb7 @derekwbrown